### PR TITLE
Handle writing file by ressource as expected by Azure

### PR DESCRIPTION
**Pull request by achretien**

### CODE REVIEW

**Comment by NiR-** [comment]
Thanks @achretien. I just saw your PR after applying the same fix as #412 to `AzureBlobStorage` and `GoogleCloudStorage` adapters. You'll be credited for the first one (my PR is now rebased ontop of yours).

**Comment by dkarlovi** [comment]
@NiR- is there an issue for this not being merged?

File: src/Gaufrette/Adapter/AzureBlobStorage.php [hunk 1]
```diff
@@ -148,13 +148,21 @@ public function write($key, $content)
...
+                }
+                $options->setBlobContentType($contentType);
             }
 
             $this->blobProxy->createBlockBlob($this->containerName, $key, $content, $options);
 
-            return Util\Size::fromContent($content);
+            if (is_resource($content)) {
+                return Util\Size::fromResource($content);
+            } else {
```
Comment (flug) [inline src/Gaufrette/Adapter/AzureBlobStorage.php]: @achretien I will remove the 'else' to return directly (the 'else' is a bit useless in your case)

**Comment by NiR-** [comment]
@dkarlovi #414 (it solves the same problem) has been merged ;)

**Comment by dkarlovi** [comment]
@NiR- yay, thanks! \o/

**Pull request closed without merging**